e2e/support/helpers/e2e-dependency-graph-helpers.ts

```diff
@@ -5,3 +5,14 @@ export const DependencyGraph = {
   selectionButton: () => cy.findByTestId("graph-selection-button"),
   dependencyPanel: () => cy.findByTestId("graph-dependency-panel"),
 };
+
+export function waitForDependencyGraph(maxRetries = 1000) {
+  if (maxRetries === 0) {
+    throw new Error("Timed out waiting for dependency graph");
+  }
+  cy.request("GET", "/api/ee/dependencies/graph/status").then(({ body }) => {
+    if (!body.dependencies_analyzed) {
+      waitForDependencyGraph(maxRetries - 1);
+    }
+  });
```

Comment:
surely we should be doing some sort of delay here before hitting the API endpoint again, slamming it 1000x times in a row with no wait is gonna put a ton of load on the server and probably actually make these tests slower since it's gonna spend so much CPU time servicing all those API requests it won't actually be able to finish the baccfill job... right??